generator/config/projection/slice.yaml

```diff
@@ -15,3 +15,41 @@ arguments:
         name: skip
         type:
             - int
+        optional: true
+tests:
+    -
+        name: 'Return an Array with Its First 3 Elements'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/projection/slice/#return-an-array-with-its-first-3-elements'
+        pipeline:
+            -
+                $project:
+                    comments:
+                        # Example uses the short form, the builder always generated the verbose form
+                        # $slice: 3
+                        $slice: [3]
```

Comment:
Looks like we will need a custom encoder. 

`$slice: [3]` returns a server error.

> [MongoServerError:](https://github.com/mongodb/mongo-php-builder/pull/54#) Invalid $slice syntax. The given syntax { $slice: [ 3 ] } did not match the find() syntax because :: Location31272: $slice array argument should be of form [skip, limit] :: The given syntax did not match the expression $slice syntax. :: caused by :: Expression $slice takes at least 2 arguments, and at most 3, but 1 were passed in.

Reply:
> the version with 3 arguments is not documented

I apologize for the headache, but that sounds like it warrants a DOCSP ticket. And make sure to tell them you shouldn't be used as the technical reviewer when/if it actually gets implemented (this frequently happens to me when I report server docs issues). It should really be someone from server, since they'll have to confirm what it actually does.

And if it should remain undocumented, server should consider revising this error message.